shotgun_api3/shotgun.py

```diff
@@ -3938,11 +3938,10 @@ def _make_call(self, verb, path, body, headers):
                 if attempt == max_rpc_attempts:
                     LOG.debug("Request failed.  Giving up after %d attempts." % attempt)
                     raise
-            except Exception:
+            except Exception as e:
                 self._close_connection()
-                if attempt == max_rpc_attempts:
-                    LOG.debug("Request failed.  Giving up after %d attempts." % attempt)
-                    raise
+                LOG.debug(f"Request failed.  Reason: {e}")
```

Comment:
Let's get the traceback since it's a debug log so we'll get more context

```suggestion
                LOG.debug(f"Request failed.  Reason: {e}", exc_info=True)
```